topic #4 Generic vs Convenience approach for Spatial data APIs/README.md

```diff
@@ -1,2 +1,281 @@
+
+### Usage example
+
+The following code fragment shows the minimum requirements for creating the engine and mounting it inside an existing router.
+
+```go
+func EnableOGCAPIFeatures() {
+  engine := apif.NewSimpleEngine(mountingPath)
+  apif.AddBaseJSONTemplates(engine)
+  apif.AddBaseHTMLTemplates(engine)
+
+  featuredatasource := myDatasourceImplementation.NewImplementation()
+  apif.EnableFeatures(engine, featuredatasource)
+  apif.AddFeaturesJSONTemplates(engine)
+  apif.AddFeaturesHTMLTemplates(engine)
+
+  engine.RebuildOpenAPI()
+  router.HandleFunc(regexp.MustCompile("^"+mountingPath), engine.HTTPHandler)
+}
+```
+
+In order to provide a datasource implementation for features, the following methods must be implemented:
+
+```go
+type FeatureService interface {
+  Collections() []featuremodels.Collection
+  Collection(string) featuremodels.Collection
+  Features(*http.Request, *featuremodels.FeaturesParams) featuremodels.Features
+  Feature(string, string) *featuremodels.Feature
+  BuildOpenAPISpecification(builder coreservices.OpenAPIBuilder)
+}
+```
+
+The readme of the library contains a code fragment that implements the datasource. The readme can be found at https://github.com/purple-polar-bear/go-ogc-api
+
+## Results
+This chapter contains the results from our research.
+
+### Library
+Before we dive into the benefits and the cost of an additional OGC API Features implementation, we will look into the results of the library development.
+
+#### Code review on library
+
+The design of the library as well as the implementation have been submitted to the project owners of Goaf (a team of Kadaster). They were asked to criticize the design. They find the implementation, at face value, sufficient for further exploration. They want to use this library for a different project. The have also stated that a conclusion cannot be drawn without actual using the code in a different project.
+
+#### Ease of expansion
+
+With respect to ease of expansion, there is decoupling in the following functional areas:
+- The 'Features' part is independent from the main engine. This means one doesn't have to use/ load the features part, but can use instead other OGC API Features, such as the tileset. This leads to ease of expansion with respect to the OGC API standards.
+- The library contains default json and html templates. However, these templates can be replaced by other json or html templates. This leads to ease of expansion with respect to the (most) common templates.
+- A developer can easily add other template formats, such as Geopackage. This can be done by simple adding other templates to a service. This leads to ease of expansion with respect to output formats.
+
+#### Subjective experience of library development
+When implementing the specifications in the library, the developers had the following experience:
+- Experience in working with Geodata, including implementing the OGC WFS 2.0 standard.
```

Comment:
zou zeggen op z'n minst de OGC API Features
(WFS 2.0 kan mogelijk zelfs in de weg zitten m.b.t. het "bedenken" van solutions)

Reply:
Ik snap je opmerking niet goed. Ik probeer hier de uitgangspositie aan het begin van het topic te benoemen; toen hadden we geen kennis van OGC API Features (wel van WFS20). Kan je uitleggen hoe ik dit anders kan doen/ hoe dit bij jou overkomt?